### PR TITLE
build(matrix): use fluid-tsc

### PR DESCRIPTION
**Pull request by DLehenbauer**
@anthony-murphy, @Abe27342 - A proper ESM package requires includes file extensions when importing files.  These were added automatically via 'ts2esm'.

### CODE REVIEW

File: packages/dds/matrix/package.json [hunk 1]
```diff
@@ -11,74 +11,66 @@
...
 	"types": "dist/index.d.ts",
 	"scripts": {
 		"api": "fluid-build . --task api",
-		"api-extractor:commonjs": "api-extractor run --local",
-		"api-extractor:esnext": "api-extractor run --config ./api-extractor-esm.json",
+		"api-extractor:commonjs": "api-extractor run --config ./api-extractor-cjs.json",
+		"api-extractor:esnext": "api-extractor run --local",
 		"bench": "cd bench && node --expose-gc -r ts-node/register src/index.ts",
-		"bench:profile": "cd bench && tsc && node -r ts-node/register --prof --logfile=profile.log --no-logfile-per-isolate src/index.ts --runInBand && node --prof-process profile.log > profile.txt && rimraf profile.log && echo See results in bench/profile.txt",
+		"bench:profile": "cd bench && node -r ts-node/register --prof --logfile=profile.log --no-logfile-per-isolate src/index.ts --runInBand && node --prof-process profile.log > profile.txt && rimraf profile.log && echo See results in bench/profile.txt",
```
Comment (DLehenbauer) [inline packages/dds/matrix/package.json:64]: Having 'tsc' in the 'bench:profile' script caused policy-check to panic due to missing task dependencies.

File: packages/dds/matrix/src/test/dirname.cjs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ */
+
+// Problem:
+//   - `__dirname` is not defined in ESM
+//   - `import.meta.url` is not defined in CJS
+// Solution:
+//   - Export '__dirname' from a .cjs file in the same directory.
+//
+// Note that *.cjs files are always CommonJS, but can be imported from ESM.
+exports._dirname = __dirname;
```
Comment (DLehenbauer) [inline packages/dds/matrix/src/test/dirname.cjs:13]: Workaround for lack of __dirname in ESM

File: packages/dds/matrix/src/test/matrix.fuzz.spec.ts [hunk 3]
```diff
@@ -229,7 +231,7 @@ describe("Matrix fuzz tests", function () {
 			clientAddProbability: 0.1,
 		},
 		reconnectProbability: 0,
-		saveFailures: { directory: path.join(__dirname, "../../src/test/results") },
+		saveFailures: { directory: path.join(_dirname, "../../src/test/results") },
```
Comment (DLehenbauer) [inline packages/dds/matrix/src/test/matrix.fuzz.spec.ts:234]: See 'dirname.cjs'

**Comment by Abe27342** [comment]
matrix source file changes look good to me. If you want to add more context on intent of this change (I assume some infra normalization as part of our esm effort, but I'm not sure right now what our end goal is) I can review the rest as well.

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**